### ISSUE TITLE
WGAN-GP Keras tutorial is slow to start the first epoch

### ISSUE THREAD

**Issue by romanovzky** [tensorflow/tensorflow#48356]
<em>Please make sure that this is an issue related to performance of TensorFlow.
As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:performance_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google colab and Arch Linux
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Binary in both cases
- TensorFlow version (use command below): 2.4.1 in both cases
- Python version: 3.9 on my computer, whatever google colab uses
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.2.1/8.1.0.77 locally, whatever google colab uses
- GPU model and memory: GeForce 1080, whatever google colab uses

**Describe the current behavior**

When I run the  [WGAN-GP tutorial from the Keras website](https://keras.io/examples/generative/wgan_gp/), I notice that the first epoch takes a really long time to start in both CPU and GPU, up to a minute.

**Describe the expected behavior**

The first epoch should start promptly, as it happens with the other tutorials. For example the [VAE](https://keras.io/examples/generative/vae/) and traditional [DCGAN](https://keras.io/examples/generative/dcgan_overriding_train_step/)

**Standalone code to reproduce the issue**

The [WGAN-GP colab](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/generative/ipynb/wgan_gp.ipynb). Choose a runtime accelerator and click on run all. The first epoch will always have a delay when starting training.

The [VAE colab](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/generative/ipynb/vae.ipynb) and [DCGAN colab](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/generative/ipynb/dcgan_overriding_train_step.ipynb) do not exhibit this regression.

**Other info**

I can reproduce the same performance issue on my local computer, so this is not a colab issue.

**Comment by tilakrayal** [comment]
Was able to reproduce the issue with TF v2.4, TF v2.5.0rc0 and TF-nightly. Please find the [gist](https://colab.research.google.com/gist/tilakrayal/01e5343d5c68634a3d9e38b436c46615/48356.ipynb) of it here. Thanks!

**Comment by jvishnuvardhan** [comment]
@romanovzky I can reproduce the issue on my local but May be it is due to the size of the model (WGAN-GP). Thanks!

**Comment by romanovzky** [comment]
@jvishnuvardhan thanks for checking this out. I have no doubt that it might be not a bug, but just how heavy the model is. However, an equivalent model (also with gradient penalty) in pytorch is fast to start, so I'd just wanted to call the attention to this "papercut" from a user perspective (my workflow is TF, but my group has a growing interest in alternative frameworks and papercuts like this can easily shift the group's preference away from TF).

**Comment by jvishnuvardhan** [comment]
@tomerk Any insights on the root-cause of "slow-start"? Thanks!

**Comment by jvishnuvardhan** [comment]
@romanovzky As this is more related to keras-team/keras, I will close this issue here. Let's follow the progress there in keras-team/keras https://github.com/keras-team/keras/issues/14580. Thanks!
